### PR TITLE
Update repo structure HDInsights

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Move dataplane libraries to new structure
Uodate project Properties

### CODE REVIEW

**Comment by chidozieononiwu** [comment]
cc @idear1203

File: sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Microsoft.Azure.HDInsight.Job.csproj [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
-    <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
-    <!-- Please do not move/edit code below this line -->
-
     <PropertyGroup>
         <PackageId>Microsoft.Azure.HDInsight.Job</PackageId>
         <Description>Azure HDInsight Job SDK Library</Description>
         <AssemblyName>Microsoft.Azure.HDInsight.Job</AssemblyName>
         <Version>3.0.0-preview</Version>
         <PackageTags>Microsoft Azure HDInsight Job;HDInsight;HDInsight Job</PackageTags>
+        <GenerateAssemblyConfigurationAttribute>false</GenerateAssemblyConfigurationAttribute>
```
Comment (weshaggard) [inline sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Microsoft.Azure.HDInsight.Job.csproj]: Why did you need to set these? If you are converting this to be like the other data-plane libraries you should instead removed the AssemblyInfo.cs file and let the build pipeline generate it now. Please verify that they same versions exist in the generated version as the one that was checked in.

File: sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Microsoft.Azure.HDInsight.Job.csproj [hunk 2]
```diff
@@ -1,23 +1,29 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
     <!-- Please do not move/edit code below this line -->
```
Comment (weshaggard) [inline sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Microsoft.Azure.HDInsight.Job.csproj]: You can remove the above line.

File: sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Microsoft.Azure.HDInsight.Job.csproj [hunk 3]
```diff
@@ -1,23 +1,29 @@
...
-    <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
-    <!-- Please do not move/edit code below this line -->
-
     <PropertyGroup>
         <PackageId>Microsoft.Azure.HDInsight.Job</PackageId>
```
Comment (weshaggard) [inline sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Microsoft.Azure.HDInsight.Job.csproj]: PackageId and AssemblyName properties are not necessary as they are redundant and can be removed from this project.

File: sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Microsoft.Azure.HDInsight.Job.csproj (hunk 1 above)
Comment (weshaggard) [reply sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Microsoft.Azure.HDInsight.Job.csproj]: For reference you can see the changes I made when I switch KeyVault over to using the data-plane targets. https://github.com/Azure/azure-sdk-for-net/pull/5149

File: sdk/hdinsight/Microsoft.Azure.HDInsight.Job/AzSdk.RP.props [hunk 4]
```diff
@@ -1,7 +1,7 @@
-<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <!--This file and it's contents are updated at build time moving or editing might result in build failure. Take due deligence while editing this file-->
-  <PropertyGroup>
-    <AzureApiTag>HDInsightJobManagementClient_2018-11-01-preview;</AzureApiTag>
-    <PackageTags>$(PackageTags);$(CommonTags);$(AzureApiTag);</PackageTags>
-  </PropertyGroup>
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (weshaggard) [inline sdk/hdinsight/Microsoft.Azure.HDInsight.Job/AzSdk.RP.props]: I think we should just remove this props file and move it into the project file.  I don't see any why this should be separate. 

@shahabhijeet the comment in this file seems to indicate that there is something that mutates this file are you aware of anything that does that?

File: sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Microsoft.Azure.HDInsight.Job.csproj [hunk 5]
```diff
@@ -0,0 +1,17 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+    <PropertyGroup>
```
Comment (weshaggard) [inline sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Microsoft.Azure.HDInsight.Job.csproj]: Can you please convert this to spaces with 2 spaces per indention like the other xml project files?

File: sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Microsoft.Azure.HDInsight.Job.csproj [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+        <PackageTags>$(PackageTags);$(CommonTags);$(AzureApiTag);</PackageTags>
+        <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+        <VersionPrefix>3.0.0</VersionPrefix>
+    </PropertyGroup>
+    
+    <ItemGroup>
+        <PackageReference Include="WindowsAzure.Storage" Version="8.1.4" />
+    </ItemGroup>
+    <!-- Please do not move/edit code below this line -->
+    <Import Condition=" Exists('$([MSBuild]::GetPathOfFileAbove(AzSdk.RP.props))') " Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.RP.props'))" />
```
Comment (weshaggard) [inline sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Microsoft.Azure.HDInsight.Job.csproj]: I assume this import should no longer be needed. Can you please verify and remove if it isn't?

File: sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Customizations/HDInsightJobManagementClient.Customization.cs [hunk 7]
```diff
@@ -39,9 +39,6 @@ public static RetryPolicy HDInsightRetryPolicy
         /// Required. The cluster dns name against which the job management is
         /// to be performed.
         /// </param>
-        /// <param name='userName'>
```
Comment (weshaggard) [inline sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Customizations/HDInsightJobManagementClient.Customization.cs]: Why was this removed?

Comment (weshaggard) [reply sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Customizations/HDInsightJobManagementClient.Customization.cs]: I assume some validation failed because this parameter does not exist?

Comment (chidozieononiwu) [reply sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Customizations/HDInsightJobManagementClient.Customization.cs]: That was raising a warning. But I guess that wasn't an actual issue. I added it back.

Comment (weshaggard) [reply sdk/hdinsight/Microsoft.Azure.HDInsight.Job/src/Customizations/HDInsightJobManagementClient.Customization.cs]: warnings will turn into errors in our CI builds so you should still clean this up. I just wanted to ensure I understood why you removed it.

**Comment by chidozieononiwu** [comment]
@Azure/azure-sdk-eng

**Review by weshaggard — APPROVED** [review]
